coralproject/talk
# [CORL-2948] emit moderation actions

## What does this PR do?
This PR updates our comment moderation pipeline to distinguish between `commentActions` and `moderationActions` during the pipeline. It causes certain phases to conditionally emit a `moderationAction` field that sets the status and terminates the pipeline.

## These changes will impact:


## What changes to the GraphQL/Database Schema does this PR introduce?
none.

## Does this PR introduce any new environment variables or feature flags?
No.

## If any indexes were added, were they added to `INDEXES.md`?
n/a

## How do I test this PR?
1. On both dev, and then on this branch Create the following comments
	- [x] non rejected comments
	- [x] comment reported
	- [x] comment reported, then rejected
	- [x] comment reported, rejected, then approved
	- [x] comment reported then approved
	- [x] comment flagged for suspect word, approved
	- [x] comment flagged for suspect word, rejected
	- [x] comment flagged for suspect word
	- [x] comment flagged for banned word, rejected
	- [x] comment flagged for banned word, rejected, then approved
	- [x] comment rejected in stream by mod
	- [x]  comment rejected via reject all comments
	- [x]  comment rejected via reject all comments on this site
	- [x]  comment rejected via spam ban
	- [x] comment approved via external moderation
	- [x] comment flagged via external moderation
	- [x] comment rejected via external moderation
2. Confirm that all queues contain the same corresponding comments after creating these on each branch.

## Where any tests migrated to React Testing Library?
No.

## How do we deploy this PR?
No special considerations should be needed.
